lib/utils.js

```diff
@@ -222,3 +224,197 @@ var parseJSON = function parseJSON(data) {
 
 // Export parseJSON
 exports.parseJSON = parseJSON;
+
+/**
+ * Extracts the metadata from HTTP response header
+ *
+ * @param {object} response - HTTP response
+ * @returns {object} Mapping from meta-data keys to values
+ */
+var extractMetadataFromHeaders = function extractMetadataFromHeaders(response) {
+  var metadata = {};
+  var rawHeaderCounter = 0;
+  for(var field in response.headers) {
+    rawHeaderCounter++;
+    if (/x-ms-meta-/.test(field)) {
+      // Metadata names must adhere to the naming rules for C# identifiers, which are case-insensitive,
+      // meaning that you can set,for example, a metadata with the following form:
+      // {
+      //    applicationName: 'fast-azure-blob-storage'
+      // }
+      // In order to return the original metadata name, the header names should be read from response.rawHeaders.
+      // That is because 'https' library returns the response headers with lowercase.
+      var key = response.rawHeaders[rawHeaderCounter * 2 - 2];
+      metadata[key.substr(10)] = response.headers[field];
+    }
+  }
+
+  return metadata;
+}
+
+exports.extractMetadataFromHeaders = extractMetadataFromHeaders;
+
+/**
+ * Checks if the given value is a valid GUID
+ *
+ * A string that contains a GUID in one of the following formats (`d` represents a hexadecimal digit whose case is ignored):
+ *
+ * - 32 contiguous digits: dddddddddddddddddddddddddddddddd
+ * - Groups of 8, 4, 4, 4, and 12 digits with hyphens between the groups
+ *    dddddddd-dddd-dddd-dddd-dddddddddddd
+ *  -or-
+ *    {dddddddd-dddd-dddd-dddd-dddddddddddd}
+ * -or-
+ *    (dddddddd-dddd-dddd-dddd-dddddddddddd)
+ * -Groups of 8, 4, and 4 digits, and a subset of eight groups of 2 digits, with each group prefixed by "0x" or "0X", and separated by commas.
+ *    {0xdddddddd, 0xdddd, 0xdddd,{0xdd,0xdd,0xdd,0xdd,0xdd,0xdd,0xdd,0xdd}}
+ *
+ * @param {string} value - String value to be verified
+ * @returns {boolean} True if the value is a valid GUID
+ */
+var isValidGUID = function isValidGUID(value) {
+  // remove all embedded whitespaces
+  value = value.replace(/\s/g, '');
```

Comment:
How could we avoid creating a new string?  The variable is declared already by being an arugment to the function.  I guess we could later do
```javascript
if ( /^[0-9a-fA-F]{1,32}$/i.test(value.replace(/\s/g, '')) 
```
to avoid reassigning the variable, but wouldn't the JS engine still be allocating and creating a new anonymous string?  Let's not bog down on this.